maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -847,12 +847,33 @@ private DefaultScanResult scanDependencies()
         {
             try
             {
+                DefaultScanResult scanResult = new DefaultScanResult( Collections.EMPTY_LIST );
+                
+                List<String> dependenciesToScanList = new ArrayList( Arrays.asList( getDependenciesToScan() ) );
+                TestListResolver includedAndExcludedTests = getIncludedAndExcludedTests();
+
+                for ( MavenProject mavenProject : session.getSortedProjects() )
+                {
+                    String groupArtifactId = new StringBuilder( mavenProject.getGroupId() ).append( ":" )
```

Comment:
Pls concatenate as follows `mavenProject.getGroupId() + ":" + mavenProject.getArtifactId()`.
This is simple and `javac` would compile to StringBuilder anyway, but longer in sources.